contents/data/sql/SQL.md

```diff
@@ -0,0 +1,242 @@
+Though you can choose to just use a text file to store your data, this will not be sustainable as the size of your application (and the data stored) grows. This is why there are several dedicated database software packages available today.
+
+**Structured Query Language** (**SQL**) is a programming language that is specifically designed for managing (storing, querying and manipulating) data in a <tooltip content="relational databases store data in tables">[relational database]({{baseUrl}}/contents/data/databases/databases.htmll#database-models)</tooltip>. As mentioned in [Codecademy](https://www.codecademy.com/articles/what-is-rdbms-sql), most <tooltip content="Relational DataBase Management System">RDBMSs</tooltip> like MySQL, Oracle, SQL Server and PostgreSQL use the SQL language. However, the syntaxes used in these distributions vary slightly. These differences may be in terms of case-sensitivity, available built-in functions, custom functions, date and time formats, and so on.  
+
+SQL distributions are widely used in many small and big companies. According to the official [MySQL website](https://www.mysql.com/why-mysql/), it is used in companies like Facebook, Google, Adobe, Paytm and Zappos (though they may not be using MySQL exclusively).
+
+<box type="tip">
+  Many online resources refer to Microsoft SQL Server as SQL. So, keep that in mind before you start making conclusions about the characteristics and features of SQL, the language.
+</box>
+
+
+### How does SQL work?
+
+Let's look at what a basic **SQL query** (or "command") for retrieving information from a data table looks like:
+
+```
+SELECT column_name(s)
+FROM table_name
+WHERE condition
+GROUP BY column_name(s)
+HAVING condition
+ORDER BY column_name(s);
+```
+
+<box type="tip">
+  The uppercase words in this query are clauses or statements, and are SQL keywords.
+</box>
+
+#### Breaking down the clauses
+
+Let's understand what the above query means by using an example to make it more concrete.
+
+Suppose have the data table `Students` shown here. We want to get a list of CS courses that have more than 1 student. The output should display the Course name and number of students, and should be sorted by increasing number of students.
+
+| ID | Name  | Course |
+| -- | ----- | ------ |
+| 1  | Alex  | CS202  |
+| 2  | Bob   | MA303  |
+| 3  | Cathy | CS202  |
+| 4  | Daren | CS202  |
+| 5  | Ellie | CS101  |
+| 6  | Fred  | MA303  |
+| 7  | Gary  | CS101  |
+| 8  | Henry | CS404  |
+
+Table 1. `Students` table
+
+From the data in our table above, entries 2, 6 and 8 should not be considered in the output as they do not meet the criteria. So, the expected output is:
+
+| Course | num |
+| ------ | --- |
+| CS101  | 2   |
+| CS202  | 3   |
+
+Table 2. Expected output
+
+The corresponding query will be:
+
+```
+SELECT Course, COUNT(*) num
+FROM Students
+WHERE Course LIKE 'CS%'
```

Comment:
I personally dont like that you had to use pattern matching in your introductory example! Why not create another column with the course faculty (so its easier to write a query)

However, this is okay as well (I fear its hard to understand if your seeing it as a beginner)